Signal/src/ViewControllers/AddToBlockListViewController.m

```diff
@@ -316,20 +319,63 @@ - (void)updateBlockButtonEnabling
 
 - (void)blockedPhoneNumbersDidChange:(id)notification
 {
-    // TODO: Once we have a list of contacts, we should update it here.
+    dispatch_async(dispatch_get_main_queue(), ^{
+        _blockedPhoneNumbers = [_blockingManager blockedPhoneNumbers];
+
+        [self updateContacts];
+    });
 }
 
 - (void)signalRecipientsDidChange:(NSNotification *)notification
 {
-    [self updateContacts];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self updateContacts];
+    });
 }
 
 - (void)updateContacts
 {
-    dispatch_async(dispatch_get_main_queue(), ^{
-        self.contacts = self.contactsManager.signalContacts;
-        [self.contactsTableView reloadData];
-    });
+    OWSAssert([NSThread isMainThread]);
+
+    self.contacts = [self filteredContacts];
+    [self.contactsTableView reloadData];
+}
+
+- (BOOL)isContactBlocked:(Contact *)contact
+{
+    if (contact.parsedPhoneNumbers.count < 1) {
```

Comment:
Why does this mean that they are blocked?

Is it just a way to exclude them from the list? If so, consider adding that as a comment since it's non-obvious.